clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -740,6 +742,112 @@ SYCL::getDeviceLibraries(const Compilation &C, const llvm::Triple &TargetTriple,
   return LibraryList;
 }
 
+SmallVector<std::string, 8>
+SYCL::getDeviceLibraries(const Compilation &C, const llvm::Triple &TargetTriple,
```

Comment:
Please add a quick note/comment about what this function does and a description of the edge cases if applicable.